### ISSUE TITLE
Лайтовый Telegram клиент для флиппера

### ISSUE THREAD

**Issue by Astrrra** [flipperdevices/flipper-hackathon-moscow#25]
### Подробно опишите идею вашего проекта

Клиентское приложение, принимающее данные от кастомного Telegram клиента по Bluetooth и позволяющее общаться текстом 

### Готовый проект будет уметь...

Будет:

- Показывать список из последних N чатов
- При выборе чата показывать последние М собщений и поле ввода текста
- Получение новых сообщений и отправка сообщений с флиппера при помощи его клавиатуры
(N и M больше 3 и меньше 100)

Не будет:
- Показывать стикеры, эмодзи, картинки и прочие вложения
- Звонить
- Проигрывать войсы
- Все остальное)

### Вы капитан команды?

- [X] Yes, I am the leader of my team

### Вы готовы приехать на Хакатон в Москву?

- [X] Yes, my team will attend

### GitHub второго участника

@Vital7

### GitHub третьего участника

@SunnyCapt

**Comment by Astrrra** [comment]
Лучше?)

**Comment by anna-oake** [comment]
Ага! Идея красивая, причем реализуемая. Мы решили подготовить RPC-сообщение для связи сторонних приложений с внешним миром, поэтому сможете реализовать это дело, даже не трогая блютус.

Но понимаете ведь, что на клавиатуре Флиппера даже пробела нет?

**Comment by Astrrra** [comment]
Можно сделать так что приложение какое-то сочетание букв будет считать пробелом

Ну или свою написать но это довольно больно будет я думаю
Но в теории есть вариант использовать D-pad для выбора букв и символов по дереву, так что в прииииинципе...

**Comment by anna-oake** [comment]
Ну можете _ взять как пробел)

**Comment by Astrrra** [comment]
А, там же _ есть, можно на пробел заменять при отправке

**Comment by Astrrra** [comment]
> Ну можете _ взять как пробел)

Хах, одновременно

**Comment by skotopes** [comment]
Шрифты у нас без юникода

**Comment by anna-oake** [comment]
При желании можно и юникодовые подключить, я без проблем рендерила хирагану на Флиппере.
Но более правильным решением видится транслитерация на стороне моста.

**Comment by Astrrra** [comment]
> При желании можно и юникодовые подключить, я без проблем рендерила хирагану на Флиппере. Но более правильным решением видится транслитерация на стороне моста.

Мы так и думали сделать

**Comment by Astrrra** [comment]
> 

Так, после некоторого гуглинга я поняла что не совсем понимаю как работать с RPC, особенно конкретно в случае флиппера. Про это есть где-то в доках?

**Comment by Astrrra** [comment]
> Ага! Идея красивая, причем реализуемая. Мы решили подготовить RPC-сообщение для связи сторонних приложений с внешним миром, поэтому сможете реализовать это дело, даже не трогая блютус.
> 
> Но понимаете ведь, что на клавиатуре Флиппера даже пробела нет?

Так, после некоторого гуглинга я поняла что не совсем понимаю как работать с RPC, особенно конкретно в случае флиппера. Про это есть где-то в доках? Я не смогла найти

**Comment by anna-oake** [comment]
Пока нет.

**Comment by Astrrra** [comment]
Окей

**Comment by LionZXY** [comment]
Я буду без дела на хакатоне, смогу помочь с этим

**Comment by Astrr** [comment]
> Я буду без дела на хакатоне, смогу помочь с этим

О, будет очень неплохо, спасибо!

**Comment by Astrrra** [comment]
> Я буду без дела на хакатоне, смогу помочь с этим
 
О, это будет очень полезно, спасибо!

**Comment by anna-oake** [comment]
### Ждём ваше финальное решение

**Если вы не идёте на Хакатон**, напишите комментарий здесь и закройте issue.

**Если изменилось число участников**, напишите здесь, кого нужно исключить.

**Если вы ТОЧНО идёте на Хакатон**, каждый участник команды должен:
1. Зарегистрироваться на мероприятие [по ссылке](https://neuronspace.timepad.ru/event/1860939/)
2. Сообщить здесь, в issue, о прохождении регистрации

**Comment by SunnyCapt** [comment]
Прошел регистрацию

**Comment by Astrrra** [comment]
Зарегалась

**Comment by ezhevita** [comment]
Тоже зарегалась

**Comment by anna-oake** [comment]
Умнички, всем отправила билеты.

**Comment by Astrrra** [comment]
Тест
